File: src/main/java/com/polygloat/dtos/request/SetTranslationsDTO.java
package com.polygloat.dtos.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class SetTranslationsDTO {
    /**
     * Source full path is stored as name in entity
     */
    @NotNull
    @NotBlank
    private String sourceFullPath;

    /**
     * Map of language abbreviation -> text
     */
    private Map<String, String> translations;
}

File: src/main/java/io/tolgee/dtos/request/SetTranslationsDTO.kt
package io.tolgee.dtos.request

import org.springframework.validation.annotation.Validated
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotNull

@Validated
data class SetTranslationsDTO(
        /**
         * Key full path is stored as name in entity
         */
        @field:NotNull @field:NotBlank
        var key: String? = null,

        /**
         * Map of language abbreviation -> text
         */
        @field:NotNull
        var translations: Map<String, String?>? = null
)
